docs/source/guides/grpc.rst

```diff
@@ -0,0 +1,1382 @@
+- Seemlessly adding gRPC support to existing Bento.
+- Streaming support (currently on our roadmap).
+
+This guide will also walk your through some of the strengths and weaknesses of serving with gRPC, as well as
+recommendation on scenarios where gRPC might be a good fit.
+
+:bdg-info:`Requirements:` This guide assumes that you have basic knowledge of gRPC and protobuf. If you aren't
+familar with gRPC, you can start with gRPC `quick start guide <https://grpc.io/docs/languages/python/quickstart/>`_.
+
+.. seealso::
+
+   For quick introduction to serving with gRPC, see :ref:`Intro to BentoML <tutorial:Tutorial: Intro to BentoML>`
+
+Get started with gRPC in BentoML
+--------------------------------
+
+We will be using the example from :ref:`the quickstart<tutorial:Tutorial: Intro to BentoML>` to
+demonstrate BentoML capabilities with gRPC.
+
+Requirements
+~~~~~~~~~~~~
+
+Install BentoML with gRPC support with :pypi:`pip`:
+
+.. code-block:: bash
+
+   » pip install "bentoml[grpc]"
+
+Thats it! You can now serving your Bento with gRPC via :ref:`bentoml serve-grpc <reference/cli:serve-grpc>` without having to modify your current service definition 😃.
+
+.. code-block:: bash
+
+   » bentoml serve-grpc iris_classifier:latest --production
+
+Interact with your gRPC BentoService
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are two ways to interact with your gRPC BentoService:
+
+1. Use tools such as :github:`fullstorydev/grpcurl`, :github:`fullstorydev/grpcui`: 
+   The server requires :github:`reflection <grpc/grpc/blob/master/doc/server-reflection.md>` to be enabled for those tools to work.
+   Pass in ``--enable-reflection`` to enable reflection:
+
+   .. code-block:: bash
+
+      » bentoml serve-grpc iris_classifier:latest --production --enable-reflection
+
+   .. include:: ./snippets/grpc/docs/grpc_tools.rst
+
+   Open a different terminal and use one of the following:
+
+2. Use one of the below :ref:`client implementations <guides/grpc:Client Implementation>` to send test requests to your BentoService.
+
+.. _workspace: https://bazel.build/concepts/build-ref
+
+.. |workspace| replace:: ``WORKSPACE``
+
+.. _build: https://bazel.build/concepts/build-files
+
+.. |build| replace:: ``BUILD``
+
+Client Implementation
+~~~~~~~~~~~~~~~~~~~~~
+
+.. note::
+
+   All of the following client implementations are :github:`available on GitHub <bentoml/BentoML/tree/main/docs/source/guides/snippets/grpc/>`.
+
+:raw-html:`<br />`
+
+From another terminal, use one of the following client implementation to send request to the
+gRPC server:
+
+.. note::
+
+   There are many different ways to generate stubs and implement your own gRPC clients.
+   Please feel free to use any workflow you like to generate stubs. 
+
+   With that being said, we recommend users to use either of the following two approaches:
+
+   - Using `bazel <bazel.build>`_ to manage and isolate dependencies (recommended)
+   - A manual approach using ``protoc`` its language-specific plugins
```

Comment:
Not sure what this is recommending. Is it about how to generate the stubs from user's code base?

Reply:
I should just remove this section.